ingest/processors/token_transfer/token_transfer_processor.go

```diff
@@ -0,0 +1,927 @@
+	strictSendResult := result.Tr.MustPathPaymentStrictSendResult()
+
+	var events []*TokenTransferEvent
+	ev, err := p.generateEventsFromClaimAtoms(tx, opIndex, opSrcAcc, strictSendResult.MustSuccess().Offers)
+	if err != nil {
+		return nil, err
+	}
+	events = append(events, ev...)
+
+	// Generate one final event indicating the amount that the destination received in terms of destination asset
+	from, to := addressWrapper{account: &opSrcAcc}, addressWrapper{account: &strictSendOp.Destination}
+	finalEvent, err := p.mintOrBurnOrTransferEvent(tx, &opIndex, strictSendOp.DestAsset, from, to, amount.String64Raw(strictSendResult.DestAmount()))
+	if err != nil {
+		return nil, err
+	}
+	events = append(events, finalEvent)
+	return events, nil
+}
+
+func (p *EventsProcessor) pathPaymentStrictReceiveEvents(tx ingest.LedgerTransaction, opIndex uint32, op xdr.Operation, result xdr.OperationResult) ([]*TokenTransferEvent, error) {
+	opSrcAcc := operationSourceAccount(tx, op)
+	strictReceiveOp := op.Body.MustPathPaymentStrictReceiveOp()
+	strictReceiveResult := result.Tr.MustPathPaymentStrictReceiveResult()
+
+	var events []*TokenTransferEvent
+	ev, err := p.generateEventsFromClaimAtoms(tx, opIndex, opSrcAcc, strictReceiveResult.MustSuccess().Offers)
+	if err != nil {
+		return nil, err
+	}
+	events = append(events, ev...)
+
+	// Generate one final event indicating the amount that the destination received in terms of destination asset
+	from, to := addressWrapper{account: &opSrcAcc}, addressWrapper{account: &strictReceiveOp.Destination}
+	finalEvent, err := p.mintOrBurnOrTransferEvent(tx, &opIndex, strictReceiveOp.DestAsset, from, to, amount.String64Raw(strictReceiveOp.DestAmount))
+	if err != nil {
+		return nil, err
+	}
+	events = append(events, finalEvent)
+	return events, nil
+}
+
+/*
+This code needs to be run to compare the diffs between the changesMap and eventsMap for the (sourceAccount, XLM) combination
+This needs to be run for each operation, except INFLATION OPERATION, and when ledgerSeq <= 8
+For more details, on why this is needed, refer - https://github.com/stellar/stellar-protocol/blob/master/core/cap-0067.md#retroactively-emitting-events
+
+The maybeGenerateMintOrBurnEvents function takes in an account and an asset, but in reality, this will only be called for operationSourceAccount and strictly for XLM
+*/
+func (p *EventsProcessor) maybeGenerateMintOrBurnEventsForReconciliation(tx ingest.LedgerTransaction, opIndex uint32, changesMap, eventsMap map[balanceKey]int64, account xdr.MuxedAccount, asset xdr.Asset) []*TokenTransferEvent {
+	var events []*TokenTransferEvent
+
+	accountStr := account.ToAccountId().Address()
+	// Create the balance key for this account and XLM asset
+	key := balanceKey{holder: accountStr, asset: asset.StringCanonical()}
+
+	// Get the balance changes from both maps
+	changesBalance, existsInChanges := changesMap[key]
+	eventsBalance, existsInEvents := eventsMap[key]
```

Comment:
```suggestion
	changesBalance := changesMap[key]
	eventsBalance := eventsMap[key]
```

see comment below where I explain why the exists check is not necessary

Reply:
reworked this to remove checks